<a href="https://colab.research.google.com/github/ElofssonLab/kb8029-book/blob/main/notebooks/day02-discussion-3.ipynb" style="display:inline-block;padding:10px 18px;background-color:#F9AB00;color:#000000;font-weight:bold;text-decoration:none;border-radius:6px;font-family:sans-serif;font-size:14px;">&#9654;&nbsp; Open in Google Colab</a>

# Day 2 — In-class discussion problem (3 of 3)

Work this out **by hand in your group first** — then run the code cell
to check your answer before presenting.

## Is a strong correlation from 5 data points trustworthy?

A group measures (standardized) length and GC content for just 5
genes:

```
x = [1.83, -3.08, 0.96, 0.07, 1.32]
y = [1.14, -0.44, 0.50, -0.27, 1.01]
```

`scipy.stats.pearsonr` gives r ≈ 0.85 — a strong correlation.

**As a group:**

1. Would you call r = 0.85 convincing evidence of a real relationship?
2. Check the p-value. Is it below the conventional 0.05 threshold?
3. Predict: if the *same underlying relationship* were measured on 50
   genes instead of 5 (same r, just less noisy per-point), would the
   p-value change? If so, in which direction, and why?

Run the cell below to check your answers.

In [1]:
import numpy as np
from scipy import stats

x5 = np.array([1.83, -3.08, 0.96, 0.07, 1.32])
y5 = np.array([1.14, -0.44, 0.50, -0.27, 1.01])
r5, p5 = stats.pearsonr(x5, y5)
print(f"n=5:  r={r5:.3f}  p={p5:.4f}")

# same underlying pattern, repeated 10x with a little added noise -> n=50
rng = np.random.default_rng(0)
x50 = np.tile(x5, 10) + rng.normal(scale=0.05, size=50)
y50 = np.tile(y5, 10) + rng.normal(scale=0.05, size=50)
r50, p50 = stats.pearsonr(x50, y50)
print(f"n=50: r={r50:.3f}  p={p50:.2e}")

n=5:  r=0.853  p=0.0663
n=50: r=0.843  p=1.50e-14


**Discussion point:** with only 5 points, r ≈ 0.85 but p ≈ 0.066 —
*not* significant at the conventional 0.05 threshold, despite looking
like a strong correlation. With the same underlying relationship
measured on 50 points instead, r barely changes (≈0.84) but p drops to
about 10⁻¹⁴ — overwhelming significance. The correlation's *strength*
barely moved; what changed is how confident you can be that it isn't
just noise. A high r from very few points is the situation
where you should be most skeptical, not least — small samples make it
easy to stumble onto an impressive-looking r by chance alone.